app/questionnaire/navigation.py

```diff
@@ -40,14 +40,25 @@ def build_navigation(self, current_group_id, current_group_instance):
             logger.debug("Building frontend navigation for group %s", group['id'])
 
             repeating_rule = SchemaHelper.get_repeat_rule(group)
+            skip_group = self._should_skip_group(current_group_instance, group)
+
+            if not skip_group:
+                if repeating_rule:
+                    navigation += self._build_repeating_navigation(repeating_rule, group, current_group_id,
```

Comment:
The repeating element will return as an array so we need += rather then append
[1, 2, 3] += [4] = [1, 2, 3, 4]
[1, 2, 3].append([4]) = [1, 2, 3, [4]]